var/spack/repos/builtin/packages/unifyfs/package.py

```diff
@@ -16,55 +16,50 @@ class Unifyfs(AutotoolsPackage):
 
     homepage = "https://github.com/LLNL/UnifyFS"
     git      = "https://github.com/LLNL/UnifyFS.git"
-    url      = "https://github.com/LLNL/UnifyFS/releases/download/v0.9.0/unifyfs-0.9.0.tar.gz"
+    url      = "https://github.com/LLNL/UnifyFS/releases/download/v0.9.2/unifyfs-0.9.2.tar.gz"
     maintainers = ['CamStan']
 
     version('develop', branch='dev')
+    version('0.9.2', sha256='7046625dc0677535f5d960187cb2e2d58a6f8cfb4dc6a3604f825257eb0891aa')
     version('0.9.1', sha256='2498a859cfa4961356fdf5c4c17e3afc3de7e034ad013b8c7145a622ef6199a0')
-    version('0.9.0', sha256='e6c73e22ef1c23f3141646aa17058b69c1c4e526886771f8fe982da924265b0f')
+    version('0.9.0', sha256='e6c73e22ef1c23f3141646aa17058b69c1c4e526886771f8fe982da924265b0f', deprecated=True)
 
     variant('auto-mount', default='True', description='Enable automatic mount/unmount in MPI_Init/Finalize')
     variant('hdf5', default='False', description='Build with parallel HDF5 (install with `^hdf5~mpi` for serial)')
     variant('fortran', default='False', description='Build with gfortran support')
-    variant('mdhim', default='False', description='Enable MDHIM build options')
     variant('pmi', default='False', description='Enable PMI2 build options')
     variant('pmix', default='False', description='Enable PMIx build options')
     variant('spath', default='True', description='Use spath library to normalize relative paths')
 
-    depends_on('autoconf',  type='build')
-    depends_on('automake',  type='build')
-    depends_on('libtool',   type='build')
-    depends_on('m4',        type='build')
-    depends_on('pkgconfig', type='build')
+    depends_on('autoconf',        type='build')
+    depends_on('automake',        type='build')
+    depends_on('automake@1.15:',  type='build', when='@0.9.2:')
+    depends_on('libtool',         type='build')
+    depends_on('m4',              type='build')
+    depends_on('pkgconfig',       type='build')
 
     # Required dependencies
-    depends_on('gotcha@1.0.3:', when='@0.9.1:')
-    depends_on('mochi-margo@0.4.3')
-    depends_on('mercury@1.0.1+bmi+sm')
+    depends_on('gotcha@1.0.3:')
```

Comment:
(question) Should this still be

```
depends_on('gotcha@1.0.3:', when='@0.9.1:')
```

now that version `0.9.0` is available again?

Reply:
Version `0.9.0` has `deprecated=True` now. Do I need to write the Spack package to still support deprecated versions? If so, I'll need to add a few other things back as well.

Reply:
(I need to check internally on this to establish a precedent)

I think if an old version is kept around, it should be buildable, even if it is `deprecated=True` (the only effect this has is to print warnings when using the deprecated version - Spack will still try to build it). If it is no longer buildable, then it should be removed.

Removing the version is not ideal in general because existing installations of the package may have issues, but if you think everyone would want to go through the effort of replacing old versions with new versions, then removing the version entirely makes sense.

Reply:
I'd be surprised if anyone is using v0.9.0 since we're still a young project and have changed a lot since that version.

That said, after looking at the Spack docs again, you are definitely correct that if the version is deprecated, it should still be buildable.
Pulling out version 0.9.0 would certainly simplify our package a bit, but I'm happy to add back what is needed to build v0.9.0 for now since this will be the first iteration of our package having that version deprecated.

Reply:
> Pulling out version 0.9.0 would certainly simplify our package a bit, but I'm happy to add back what is needed to build v0.9.0 for now since this will be the first iteration of our package having that version deprecated.

Absent other considerations we generally recommend keeping old versions, but in this case I think it makes sense to remove the version.

Reply:
Sounds good. I'll remove it.

That said, for future updates (unless I've discussed it and have a really good reason to remove) I'll know to keep them buildable and deprecate them first.